### PR TITLE
Convert some auth middleware to async functions

### PR DESCRIPTION
**Pull request by nwalters512**
I've been running Node 21 in dev to pick up on potential changes we'll need to handle, and I found one: Node now logs every time a promise-returning function is passed to `util.promisify`. This shows up in one main place for us: we sometimes want to use middleware as normal functions, so we'll pass them to `util.promisify`. However, those middleware are often wrapped in `express-async-handler`, which does indeed turn them into functions that return promises.

Rather than always running these functions through `util.promisify`, it seemed to make more sense to me to make the underlying functionality available through normal exported functions. That's what this PR does.

### CODE REVIEW

File: apps/prairielearn/src/middlewares/authzWorkspace.js [hunk 1]
```diff
@@ -44,8 +42,11 @@ module.exports = asyncHandler(async (req, res, next) => {
     await authzCourseOrInstance(req, res);
 
     if (isEnterprise()) {
-      const checkPlanGrants = promisify(require('../ee/middlewares/checkPlanGrants').default);
-      await checkPlanGrants(req, res);
+      const hasPlanGrants = await checkPlanGrantsForLocals(res.locals);
+      if (!hasPlanGrants) {
+        res.redirect(`/pl/course_instance/${res.locals.course_instance.id}/upgrade`);
+        return;
+      }
```
Comment (nwalters512) [inline apps/prairielearn/src/middlewares/authzWorkspace.js:49]: This was actually subtly broken before. The middleware in use before will redirect if the user doesn't have the necessary plan grants, but we had no way of knowing if it redirected, so we'd just continue one processing the request and would probably eventually fail with "you can't set headers again" when we tried to respond later on.

**Comment by nwalters512** [comment]
Kicking this back into draft while I work through some TypeScript issues.

File: apps/prairielearn/src/middlewares/selectAndAuthzInstanceQuestion.js [hunk 2]
```diff
@@ -59,6 +59,10 @@ const middleware = asyncHandler(async (req, res, next) => {
       }
     }
   }
+}
+
+const middleware = asyncHandler(async (req, res, next) => {
```
Comment (mwest1066) [inline apps/prairielearn/src/middlewares/selectAndAuthzInstanceQuestion.js:64]: This doesn't seem be exported and it looks different to all the other handlers?

**Review by mwest1066 — APPROVED** [review]

File: apps/prairielearn/src/middlewares/selectAndAuthzInstanceQuestion.js (hunk 2 above)
Comment (nwalters512) [reply apps/prairielearn/src/middlewares/selectAndAuthzInstanceQuestion.js:64]: It is exported on line 69, but I agree that this should be made consistent with the other files (that is, we should use `export default asyncHandler`. Will change.

Comment (nwalters512) [reply apps/prairielearn/src/middlewares/selectAndAuthzInstanceQuestion.js]: Changed in 48dcfa1.

Comment (mwest1066) [reply apps/prairielearn/src/middlewares/selectAndAuthzInstanceQuestion.js]: Thanks!

**Pull request merged**